Packages/com.unity.inputsystem/Tests/InputSystem/CoreTests_Remoting.cs

```diff
@@ -128,23 +134,40 @@ public void Remote_CanConnectInputSystemsOverEditorPlayerConnection()
         // Bind a local remote on the player side.
         var local = new InputRemoting(InputSystem.s_Manager);
         local.Subscribe(connectionToEditor);
-        local.StartSending();
 
-        connectionToPlayer.Subscribe(observer);
+        connectionToEditor.Subscribe(local);
+        connectionToPlayer.Subscribe(observerEditor);
+        connectionToEditor.Subscribe(observerPlayer);
+
+        fakeEditorConnection.Send(RemoteInputPlayerConnection.kStartSendingMsg, null);
 
         var device = InputSystem.AddDevice("Gamepad");
         InputSystem.QueueStateEvent(device, new GamepadState());
         InputSystem.Update();
         InputSystem.RemoveDevice(device);
 
-        ////TODO: make sure that we also get the connection sequence right and send our initial layouts and devices
-        Assert.That(observer.messages, Has.Count.EqualTo(4));
-        Assert.That(observer.messages[0].type, Is.EqualTo(InputRemoting.MessageType.Connect));
-        Assert.That(observer.messages[1].type, Is.EqualTo(InputRemoting.MessageType.NewDevice));
-        Assert.That(observer.messages[2].type, Is.EqualTo(InputRemoting.MessageType.NewEvents));
-        Assert.That(observer.messages[3].type, Is.EqualTo(InputRemoting.MessageType.RemoveDevice));
+        fakeEditorConnection.Send(RemoteInputPlayerConnection.kStopSendingMsg, null);
+
+        // We should not obseve any messages for these, as we stopped sending!
+        device = InputSystem.AddDevice("Gamepad");
```

Comment:
Preferably use `InputSystem.AddDevice<Gamepad>();`. The `AddDevice("Gamepad")` thing that's still in some tests is mainly a leftover.